core/src/main/java/org/elasticsearch/index/query/NestedQueryBuilder.java

```diff
@@ -88,4 +138,89 @@ protected void doXContent(XContentBuilder builder, Params params) throws IOExcep
     public final String getWriteableName() {
         return NAME;
     }
+
+    @Override
+    protected boolean doEquals(NestedQueryBuilder that) {
+        return Objects.equals(query, that.query)
+                && Objects.equals(path, that.path)
+                && Objects.equals(scoreMode, that.scoreMode)
+                && Objects.equals(queryInnerHits, that.queryInnerHits);
+    }
+
+    @Override
+    protected int doHashCode() {
+        return Objects.hash(query, path, scoreMode, queryInnerHits);
+    }
+
+    public NestedQueryBuilder(StreamInput in) throws IOException {
```

Comment:
just because it seems like it's used only internally, that's it. No strong opinions though ;)